File: android/java/com/wrapper/payload/FirebaseDynamicLinks.java
package com.wrapper.payload;

import android.content.Intent;
import android.app.Activity;
import android.net.Uri;
import androidx.core.app.ShareCompat;
import android.util.Log;

public class FirebaseDynamicLinks {
    private static String TAG = "FirebaseDynamicLinks";
    private static Activity sActivity;
    private static int REQUEST_INVITE = 25;

    public static void setActivity(Activity activity) {
        sActivity = activity;
    }

    public static void shareLink(String linkStr) {
        if (linkStr == null) {
            return;
        }
        Log.d(TAG, "shareLink:" + linkStr);
        Intent shareIntent = ShareCompat.IntentBuilder.from(sActivity)
                                                    .setType("text/plain")
                                                    .setText(linkStr).getIntent();
        sActivity.startActivity(shareIntent);
    }
}

File: android/cpp/FirebaseDynamicLinks.cpp
#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
#include "FirebaseDynamicLinks.h"
#include "platform/android/jni/JniHelper.h"
#include <string>

void FirebaseDynamicLinks::shareLink(std::string linkStr) {
    cocos2d::JniHelper::callStaticVoidMethod(
            "com/wrapper/payload/FirebaseDynamicLinks",
            "shareLink", linkStr);
}
#endif

